canonical/vanilla-framework
# Update strip component to new theming system

## Done

- Updates strips to use new theming system with `is-light` `is-dark` `is-paper` class names.
- Introduces new `p-strip--highlighted` to replace a variety of strip colour variants with a single themed version of a strip with an alternative backround colour
- Deprecates a number of old strips that are not meant for new designs: `p-strip--dark` `p-strip--light` `p-strip--white` `p-strip--image` `p-strip--suru`, etc. They should be replaced by new themed versions.
- Updates all the necessary docs and examples.

Fixes [WD-8473](https://warthogs.atlassian.net/browse/WD-8473)

## QA

- Open [demo](https://vanilla-framework-4996.demos.haus/docs/patterns/strip)
- Check the dark strip example: https://vanilla-framework-4996.demos.haus/docs/examples/patterns/strips/dark
  - verify in inspector that it takes colors from CSS variables
  - change the theme to `is-light`, `is-paper` to check if it adjusts
- Check the highlighted strip example: https://vanilla-framework-4996.demos.haus/docs/examples/patterns/strips/highlighted
  - change the theme class (on body or on strip)
  - make sure the colour changes: on dark theme it should use a "lighter" dark, on paper it should use white, on light it should use light grey
- Verify that old deprecated versions are still supported
  - [white](https://vanilla-framework-4996.demos.haus/docs/examples/patterns/strips/white)
  - [suru](https://vanilla-framework-4996.demos.haus/docs/examples/patterns/strips/suru)
  - [light](https://vanilla-framework-4996.demos.haus/docs/examples/patterns/strips/strips-light)
  - [dark](https://vanilla-framework-4996.demos.haus/docs/examples/patterns/strips/strips-dark)
  - [bordered](https://vanilla-framework-4996.demos.haus/docs/examples/patterns/strips/is-bordered)
  - [image](https://vanilla-framework-4996.demos.haus/docs/examples/patterns/strips/image)
  - [accent](https://vanilla-framework-4996.demos.haus/docs/examples/patterns/strips/accent)
- Review updated documentation:
  - https://vanilla-framework-4996.demos.haus/docs/patterns/strip

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)


## Screenshots

<img width="1420" alt="image" src="https://github.com/canonical/vanilla-framework/assets/83575/5c867ff2-0925-4e2b-908c-78d8fca849c2">



[WD-8473]: https://warthogs.atlassian.net/browse/WD-8473?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
